mypy/types.py

```diff
@@ -1602,7 +1599,7 @@ def copy_modified(
             variables=variables if variables is not _dummy else self.variables,
         )
 
-    # the following are copied from CallableType. Is there a way to decrease code duplication?
+    # TODO: here is a lot of code duplication with Callable type, fix this.
```

Comment:
I guess I should write this historical note here in case it helps or something: I originally wanted to make every Callable have a Parameters attribute, meaning it doesn't hold its own args anymore. That seemed like too much work though! ... maybe these can inherit instead?

Reply:
There are three possible ways to fix this:
* Delete `Parameters` and use `CallableType` instead. (But I don't like this way, because it would be a hack.)
* Make `arg_types` etc. properties on `CallableType` that forward values from its `parameters` attribute. This way I guess it will be much less work, but may have visible performance impact.
* Indeed, both can inherit these methods from say `ArgumentHelpers` mixin.

Ideally I think we should try second option, measure performance, if it is OK, go with it, if it is bad, try option 3.